Dockerfile

```diff
@@ -0,0 +1,42 @@
+################################################################################################################ builder
+
+FROM maven:3-jdk-8 as builder
+WORKDIR /build
+
+# this block will make docker cache the dependencies for subsequent builds of the same image.
+COPY pom.xml .
+COPY subsonic-rest-api/pom.xml ./subsonic-rest-api/pom.xml
+COPY integration-test/pom.xml ./integration-test/pom.xml
+COPY airsonic-sonos-api/pom.xml ./airsonic-sonos-api/pom.xml
+COPY airsonic-main/pom.xml ./airsonic-main/pom.xml
+RUN mvn de.qaware.maven:go-offline-maven-plugin:resolve-dependencies
+
+COPY . .
+RUN mvn package
+
+################################################################################################################ runtime
+
+FROM openjdk:8-jre-alpine as runtime
+LABEL description="Airsonic is a free, web-based media streamer, providing ubiquitious access to your music." \
+      url="https://github.com/airsonic/airsonic"
+ENV AIRSONIC_PORT=4040 AIRSONIC_DIR=/airsonic CONTEXT_PATH=/ UPNP_PORT=4041 JVM_HEAP=256m
```

Comment:
```suggestion
ENV AIRSONIC_DIR=/airsonic
ENV CONTEXT_PATH=/
ENV JVM_HEAP=256m
```

Reply:
agree with the removal of unnecessary indirections (ports). Sure about the multiple lines? Each of them will commit a new layer to the docker image.

Reply:
They won't.
ENVs don't commit new layers to docker image.

(Additionally I don't think the ENVs should be removed, see my comment below)

Reply:
> ENVs don't commit new layers to docker image.

They do not add any size to the image, but they do add layers:

```
airsonic (feature/docker-build ✘)✹ ᐅ docker history 1e9339ec6607
IMAGE               CREATED              CREATED BY                                      SIZE                COMMENT
(...)
f33cbef52b9b        About a minute ago   /bin/sh -c #(nop)  ENV JVM_HEAP=256m            0B
69fc464b7099        About a minute ago   /bin/sh -c #(nop)  ENV UPNP_PORT=4041           0B
d0a91aa6ecfc        About a minute ago   /bin/sh -c #(nop)  ENV CONTEXT_PATH=/           0B
89e7e6bb414f        About a minute ago   /bin/sh -c #(nop)  ENV AIRSONIC_DIR=/airsonic   0B
0533c1173dc5        About a minute ago   /bin/sh -c #(nop)  ENV AIRSONIC_PORT=4040       0B
(...)
```

* thumbs up:  for readability and each `ENV` on a seperate line
* thumbs down: collapse `ENV` on one line.

Reply:
They're temporary intermediate images, they don't actually add layers.

Here's the official documentation:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
```
In older versions of Docker, it was important that you minimized the number of layers in your images to ensure they were performant. The following features were added to reduce this limitation:

Only the instructions RUN, COPY, ADD create layers. Other instructions create temporary intermediate images, and do not increase the size of the build.
```

Reply:
If we split into new lines and keep the variables, looks like this can be resolved.

Reply:
adapted formatting, did the same to the VOLUME statements.